server/channels/api4/hosted_customer.go

```diff
@@ -293,3 +295,34 @@ func selfHostedInvoicePDF(c *Context, w http.ResponseWriter, r *http.Request) {
 		r,
 	)
 }
+
+func handleSubscribeToNewletter(c *Context, w http.ResponseWriter, r *http.Request) {
+	const where = "Api4.handleSubscribeToNewletter"
+	cloud := c.App.Cloud()
```

Comment:
Had another thought related to Nick's thought about cloud instances. Team edition doesn't have the cloud interface, so if we want to support this flow for team edition users (which makes sense to me), we'll need to move the code that actually sends the request to CWS out from enterprise and into mattermost-server.

Reply:
Hey @johndavidlugtu and @marianunez do we want to support newsletter entries for team edition?

Reply:
@neallred atm team edition don't get the checkbox but a link to mm.com so they can subscribe

Reply:
> @neallred atm team edition don't get the checkbox but a link to mm.com so they can subscribe

This is a good solution for now. It would be nice to have this for team edition but our first priority right now would be for enterprise edition. It's not worth the effort to build out the ability to support CWS endpoints in team edition just for this feature.